docs/devguide/docs/tests.md

```diff
@@ -108,6 +108,115 @@ There are two additional items to note:
 
 If you login to the Predator UI after creating the test, you will notice that the test has been added with a type of **dsl**. 
 
+You can now run the test as you would any other. 
+
 ![Screenshot](images/dsltestinui.png)
 
-You can now run the test as you would any other. 
\ No newline at end of file
+## Creating a Test with custom logic in Javascript
+Tests can use custom Javascript functions , for using Javascript functions in your tests you should first create text file, with your Javascript code,
+and upload it to public repository. (S3, dropbox,google docs etc... ).
+when creating a test use file_url parameter to send URL path to your Javascript file .
+Predator will download and save this file (Any changes after creating test will not be reflected.)
+
+Here's an example:
+
+
+```JSON
+{
+  "name": "custom logic in Javascript example",
+  "description": "custom logic in Javascript",
+  "file_url": "https://www.dropbox.com/s/yourFilePath/fileName.txt?dl=1",
+  "type": "basic",
+  "before": {
+  	"steps": [
+  		{
+  			"action": "petstore.create-pet"
+  		}	
+  	]
+  },
+  "scenarios": [
+  	{
+  		"scenario_name": "Only get pet",
+  		"steps": [
+  			{
+  				"action": "petstore.get-pet"
+  			}	
+  		]
+  	}
+  ]
+}
+```
+To create your scenarios with Javascript function , work with Artillery guidelines.
+
+* Artillery documentation: https://artillery.io/docs/http-reference/#loading-custom-js-code
+
+Here's an example for test using JS code:
```

Comment:
Here's an example for test using custom JS code:

Reply:
fixed